### PR TITLE
Unhardcode "crater", improve appearance of mutable craters

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
Content "Mutable (the large not extra ones) craters are now round"

#### Purpose of change

Hardcoding bad
Square craters ugly

#### Describe the solution

Unhardcodes "crater" and removes all references to it in the c++
New map creates round craters with a single map for the edges using neighbour based nests
Also added a small version for some more variety, annoyingly #59781 means I can't use less bombs without resulting in straight borders on the explosion (tried 1 and 2)
All the crater maps now use fallback_predecessor_mapgen allowing the mutable to spawn over roads sensibly and resulting in items where stuff like trees and road are destroyed
Slightly reduces the occurrences to make up for the fact they're bigger now
All unsolved joins are resolved over already altered terrain so should never cause issues

#### Describe alternatives you've considered

Not ruining the beautifully simplistic mutable by making the terrain's radius small enough it wouldn't need to be changed to accommodate corners
Sadly while vehicles that spawn in the blast get decimated as expected, ones that spawn over the place_rubble get damaged very inconsistently and I'm getting the occasional crash that appears to be from that so I'm removing all vehicles outside of the small area of no rubble which basically means you'll never see vehicles getting wrecked v_v
Ideally it would be able to spawn anywhere but that's not viable presently
I might make a follow up PR where I:
- Make it more randomly patchy at the edges (see the wibbly wobbly commit for the general idea)
- Allow it to spawn over more exclusively z0 maps like fields/pastures etc
- Add sensible items to the crater, maybe a fighter crash site variant as an alternate choice with no join in the first phase so it only makes one crater

#### Testing

Compiled and checked every phase variant spawned by making them blue one at a time

<details>
  <summary>Testing Screenshot</summary>

image
  
</details>

#### Additional context

<details>
  <summary>Screenshots</summary>

image
Example over road

image
Example in forest

image
Very rare vehicle getting destroyed

image
Map screenshot to show potential sizes and shapes, minimum size is one 3x3 crater (Obv this many don't spawn)
  
</details>

### CODE REVIEW

**Comment by ashGlaw** [comment]
Presently, are craters just sheer pits? Might make some sense for some ramps to spawn, that way the craters can be explored and don't provide some handy zombie storage

**Comment by Procyonae** [comment]
> Presently, are craters just sheer pits? Might make some sense for some ramps to spawn, that way the craters can be explored and don't provide some handy zombie storage

They're currently (and still will be) explosions that occur when they enter your reality bubble the same as the extra currently which makes them fairly random without needing to do anything clever but at the same time means adding ramps would be very awkward if not impossible. Feasibly the fake explosive could be replaced with an EOC that causes an explosion and then creates ramps/slopes where appropriate but that's very much out of scope.
I could swap the explosives out for "set": "bash" or "rubble" nests with slopes at the sides but that would be messy and neither of those are particularly robust at destroying terrain, or I could just make holes and manually fill them with expected items but that's far from ideal.
I will say that they don't currently place any items themselves other than a very underwhelming "wreckage" itemgroup around the actual blast so there's no purpose to exploring them particularly but the zombie storage problem is pretty silly.
Also just generally I'd argue destroyed natural terrain should result in ramps/slopes but the current implementation of ramps/slopes is really iffy already so I defo won't be attempting to add that.

File: data/json/mapgen/crater.json [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+  {
+    "type": "palette",
+    "id": "crater_palette",
+    "terrain": {
+      ".": "t_region_groundcover_barren",
+      "?": [ [ "t_region_groundcover_barren", 3 ], "t_null" ],
+      "0": "t_region_groundcover_barren"
+    },
+    "items": { ".": { "item": "wreckage", "chance": 2 }, "?": { "item": "wreckage", "chance": 1 } },
+    "nested": { ".": { "chunks": [ [ "rubble", 10 ], [ "null", 3 ] ] }, "?": { "chunks": [ [ "rubble", 3 ], [ "null", 10 ] ] } }
```
Comment (GuardianDll) [inline data/json/mapgen/crater.json:11]: What's the point to create 1x1 nests, if you can simply 
```json
"furniture": {
      ".": [ [ "f_rubble", 10 ], [ "null", 3 ] ]
      "?": [ [ "f_rubble", 3 ], [ "null", 10 ] ]
    },
```

Comment (Procyonae) [reply data/json/mapgen/crater.json:11]: place_rubble bashes terrain/furniture which is somewhat relevant bc it's using fallback_predecessor_mapgen so there could be road/trees/street lamps etc on the ?, the current way . is set up that stuff would've already been dismantled so it'd be useless there but I might still tweak the terrain distributions anyway I'm getting the shapes how I want first

Comment (Procyonae) [reply data/json/mapgen/crater.json:11]: There is also this #66269 which might eventually lead to rubble and set bash being more appropriate than manually placing rubble for general use too

**Comment by LovamKicsiGazsii** [comment]
The screenshot's looking seriously awesome, well done so far!

**Comment by Procyonae** [comment]
Innawood's crater was added in #54149, hasn't been touched since and only appears to exist to remove the radioactivity and that's been removed from the vanilla crater anyway to save you some time ^^

**Review by Light-Wave — APPROVED** [review]

**Pull request merged**